sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,111 @@
+    
+    static var subcriptionIDReplacement = "99999999-9999-9999-9999-999999999999"
+    
+    static var standardHeaderScrubbing: [String : FilterBehavior] = [
+        "location": .closure({ key, val in
+            scrubSubscriptionId(from: val ?? "")
+        }),
+        "operation-location": .replace(""),
+        "azure-asyncoperation": .replace(""),
+        "www-authenticate": .replace(""),
+        "access_token": .replace("")
+        
+    ]
+    
+    static func scrubSubscriptionId(from string: String?) -> String? {
+        
+        guard var editedString = string else {
+            return nil
+        }
+        editedString.replaceAllMatches(usingRegex: Filter.subscriptionIDRegex, withReplacement: "$1/\(Filter.subcriptionIDReplacement)", options: .caseInsensitive)
+        editedString.replaceAllMatches(usingRegex: Filter.graphSubscriptionIDRegex, withReplacement: "$1/\(Filter.subcriptionIDReplacement)", options: .caseInsensitive)
+        
+        return editedString
+        
+        
+    }
+    
+    
+    
+    func scrubSubscriptionIDs(from request: URLRequest) -> URLRequest {
+        
+        var cleanRequest = request
+        
+        let dirtyHeaders = request.allHTTPHeaderFields ?? [:]
+        let cleanHeaders = dirtyHeaders.mapValues { val in
+            return Filter.scrubSubscriptionId(from: val) ?? val
+        }
+        
+        cleanRequest.allHTTPHeaderFields = cleanHeaders.isEmpty ? nil : cleanHeaders
+        cleanRequest.url = URL(string: Filter.scrubSubscriptionId(from: request.url?.absoluteString))
+        cleanRequest.mainDocumentURL = URL(string: Filter.scrubSubscriptionId(from: request.mainDocumentURL?.absoluteString))
+        cleanRequest.httpBody = Filter.scrubSubscriptionId(from: String(data: request.httpBody, encoding: .utf8))?.data(using: .utf8)
+
+        return cleanRequest
+    }
+    
+    func scrubSubscriptionIDs(from response: Foundation.URLResponse, data: Data?) -> (Foundation.URLResponse, Data?) {
+        guard let dirtyData = data else {
+            return (response, data)
+        }
+        let responseDataString = String(data: dirtyData, encoding: .utf8)
+        let cleanedDataString = Filter.scrubSubscriptionId(from: responseDataString)
+        let cleanedData = cleanedDataString?.data(using: .utf8)
+        return (response, cleanedData)
+        
+    }
+    
+}
+
+extension String {
+    
+    mutating func replaceAllMatches(usingRegex: String, withReplacement: String, options: NSRegularExpression.Options = [], matchingOptions: NSRegularExpression.MatchingOptions = []) {
+        do {
+            let regularExpression = try NSRegularExpression(pattern: usingRegex, options: options)
+            let range = NSRange(location: 0, length: self.utf8.count)
+            self = regularExpression.stringByReplacingMatches(in: self, options: matchingOptions, range: range, withTemplate: withReplacement)
+        } catch {
+            return
+        }
```

Comment:
I'd get rid of the `do-catch` by moving the `NSRegularExpression` initialization out of here. Otherwise, if an error is thrown, then the string stays unchanged, which isn't the intention.